pyopencl/array.py

```diff
@@ -2304,20 +2305,19 @@ def arange(queue, *args, **kwargs):
     # argument processing -----------------------------------------------------
 
     # Yuck. Thanks, numpy developers. ;)
-    from pytools import Record
+    from dataclasses import dataclass, field
 
-    class Info(Record):
-        pass
+    @dataclass
+    class _ArangeInfo:
+        start: Optional[int] = None
+        stop: Optional[int] = None
+        step: Optional[int] = None
+        dtype: Optional["np.dtype"] = None
+        allocator: Optional[Any] = None
+        wait_for: List[cl.Event] = field(default_factory=list)
```

Comment:
I think this might be better as a class-local.

Reply:
What I mean is that I would prefer to not repeatedly incur the cost for code generation in `@dataclass`.

Reply:
Ah, completely misunderstood that, sorry!

Moved it out of the function in https://github.com/inducer/pyopencl/compare/f9e31f41439c9a7fdfc0ae8476a05f4d8388e125..b0a975c4433fff5659fe55d8da3f117788bf199b.